### DISCUSSION TITLE
Auto join topics

### DISCUSSION THREAD

**Discussion by geecrumb** [balzack/databag#102] [Ideas]
I use databag for my family and would like to make new users autojoin some of the rooms/topics without the need to first connect to them (contacts) and then add them to all the topics. Or even make it so that users from my homeserver are automatically added to all the topics all the time as long as they are not secured.

**Comment by balzack** [comment]
Topics are generally owned by a specific account. I see the value in what you propose, but in my mind it would more apply to an 'admin' topic or somehow a designated topic make public for the homeserver.

I will think about a way to achieve this. Thanks for the suggestion. I see minimizing the effort it takes to onboard contacts as a huge plus. Thank you!

**Reply by geecrumb** [reply]
After using Databag a bit more with at least one person (lol - will be around 7 on my server in the next days) its not that bad as I first thought. But the "work" it needs to write with someone that already is on **your own** server (which has gotten a token from you) is somehow a bit much. Not thats it not doable, dont understand me wrong, it just "could" be somehow better.

Right now the workflow is: Gets invited (token) - registers - edits his profile (username) - clicks the button to even be found in the register - needs to connect to someone - needs to be invited to a topic.

Even for, lets say something like a discord community which uses databag now, people should somehow be able to join topics themself as long as the topic is not locked down by encryption.

**Comment by balzack** [comment]
A trivial change would be to have the user's default to 'searchable in registry', but that only takes away one of the steps.

I agree that the startup workflow is too cumbersome.

Perhaps an admin option where any new account is initially connected to all other accounts on that server, would be a good idea? I wonder if a middle ground is better, where the admin can designate a set of accounts that are automatically connected to any new account.

Perhaps within an account, a topic can be designated as auto-joinable. Every time a new contact is established they are added to the designated topic.

I lean towards those approaches, but I'm sure there are other options too. Any thoughts?

**Reply by geecrumb** [reply]
I thought about a default to the "searchable in registry" too but how you say - it doesnt make a big difference.
To connect all accounts on one server (as an admin option) would be good for such family & friend case servers and make way (for me) easier. This way I would not have to talk to older people in my family, explaining to them that I add them as a "friend", then they have to accept it and first then I can invite them to the topics :D

**Comment by square-foot** [comment]
I have just posted a piece of code for a new user to join any topics created by an "admin" user. It is an external code -- does not modify Databag itself. It is in the "Show and Tell" section. I am new to Databag. I loved the minimalist approach and security concerns it respects. I hope my code is useful

**Reply by timaschew** [reply]
And updates in this or similar ideas?

**Reply by balzack** [reply]
@timaschew I lost track of this because I never saw a PR. I see the value of an 'auto join', but I could use more information on the use cases of interest. Would it be restricted to people you are connected with? How would the topic discovery occur?
